start.js

```diff
@@ -0,0 +1,54 @@
+const http = require('http');
+const soldiers = require('./routes/soldiers.js');
+const duties = require('./routes/duties.js');
+const justiceBoard = require('./routes/justiceBoard.js');
+
+function runServer() {
+  let server = http.createServer(function (req, res) {
+    let body = [];
+    req.on('data', (chunk) => {
+      body.push(chunk);
+    }).on('end', () => {
+      body = Buffer.concat(body).toString();
+      if (body !== "") {
+        body = JSON.parse(body);
+      }
+      const parsedUrl = (req.url.split("/"))[1];
+      if ("soldiers" === parsedUrl ||
+        (parsedUrl.length > 14 && "soldiers?name=" === parsedUrl.substr(0, 14))) {
+        soldiers.handleSoldierReq(req, body, function (err, result) {
+          if (err) {
+            res.statusCode = 404;
```

Comment:
Talking about errors.. this 404 isn't correct if the user did enter the correct route.
Let the method choose the status and return it here as recieved.
(and then `return res.end(err || result)` ;)